src/Root.js

```diff
@@ -35,6 +48,13 @@ class Root extends Component {
   }
 
   addReducer = (key, reducer) => {
+    if (this.queryResourceStateKey === key) {
+      const originalReducer = reducer;
+      const initialQueryObject = queryString.parse(window.location.search);
+      // eslint-disable-next-line no-param-reassign
+      reducer = (state = Object.values(initialQueryObject).length ? initialQueryObject : undefined, action) => originalReducer(state, action);
```

Comment:
Minor quibble: please avoid mutating the parameter here, couldn't you leave `reducer` as your `originalReducer` and create a new name for the changed one?